lib/housekeeping.js

```diff
@@ -37,8 +37,42 @@ class HouseKeeper {
     this.monitor = monitor;
     this.tagger = tagger;
   }
+  
+  _calculateTotals(tabulatedVolumes) {
+    let totalVolumes = {};
+    
+    tabulatedVolumes.forEach(function(volume) {
+      if (typeof volume != 'undefined') {
+        if (volume.type in totalVolumes) {
+          totalVolumes[volume.type] = {
+            count: totalVolumes[volume.type]['count'] + 1,
+            gb: totalVolumes[volume.type]['gb'] += volume.size,
+          };
+        }
+        else {
+          totalVolumes[volume.type] = {
+            count: 1,
+            gb: volume.size,
+          };
+        }
+      }
+    });
+    
+    return totalVolumes;
+  }
 
-  async _handleVolumeData(volume) {}
+  // Tabulate the required data
+  async _handleVolumeData(volume) {
+    let volumeData = {
+      type: volume.VolumeType,
+      size: volume.Size,
+      status: volume.State,
+    };
+
+    return new Promise((resolve, reject) => {
+      resolve(volumeData);
+    });
+  }
```

Comment:
I just noticed that you suggested combining the `_handleVolumeData(...)` and `_calculateTotals(...)` functions in the "Questions" section. It's my opinion that we stick with the original approach and call the other helper methods that actually do something with the data (e.g. `_calculateTotals(...)`) from `_handleVolumeData(...)`. We can rename `_handleVolumeData(...)` to have a clearer name, if you think that would help. We can actually remove this pre-processing step from `_handleVolumeData(...)` since it doesn't actually help us at all.

Reply:
Oh my. Please forgive me for my cluelessness. We don't need a function like `_handleVolumeData(...)` at all, do we? We don't ever want a function that deals with only the individual volumes, because then we'd need to save the totals globally. It seems like we can safely remove that, right? And then `_sweepVolumes(..)` can just call `_calculateTotals(...)` and `_updateVolumesCounts(...)` directly, with no `Promise.all(...)` needed.

Reply:
So you're thinking keep the `_handleVolumeData()` as a no-op function? I wasn't sure what it means to take "a page of data from the first PR and tabulates the required data". I was thinking this part would be somewhere implemented in `_handleVolumeData()` but I wasn't sure what we are expecting to get out of it. If tabulating is just what I'm interpreting it as (just take the values we want from AWS and return them in an array), we can probably combine it in to `_calculateTotals()` (but it might not be following the single responsibility principle then).

Reply:
I don't think I see why we'd need to do any "combining", or worry about single responsibility. Currently, `_handleVolumeData(...)` does essentially nothing. It just renames some fields of the response objects and creates a new object with only those fields. We can just pass the original objects to `_calculateTotals(...)` and use the actual field names (i.e. VolumeType, Size, and State).